db/migrate/20180810162637_delete_assignments_with_duplicate_mission_and_user.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class DeleteAssignmentsWithDuplicateMissionAndUser < ActiveRecord::Migration[5.1]
+  def up
+    duplicates = []
+    dup_attrs = Assignment.select(:mission_id, :user_id).group(:mission_id, :user_id).having("count(*) > 1")
+
+    dup_attrs.each do |dup|
+      dup_assignments = Assignment.where(mission_id: dup.mission_id, user_id: dup.user_id)
+      duplicates << dup_assignments.first
+    end
+
+    duplicates.map(&:destroy)
```

Comment:
Two things here:

1. Why not do this in the loop? Then you don't need an extra instance variable and the code is simpler.
2. Only use `map` if you intend to do something with the result. Use `each` otherwise. If you're not familiar with the difference, please read up and practice. Really important methods, both!

Reply:
thanks